### PR TITLE
Issue 6348. Use not identical rather than not equal to compare config files

### LINKED ISSUE #6348: Config sync doesn't see reordering of elements in JSON file as a change
## Description of the bug

I use a staging folder that isn't emptied so it can be version controlled. When I receive config edits from other developers I run "bee bcim" to import them and update my local site. However, if the only change in the file is that the order of some fields in the view display have changed, then Bee can't detect that change and therefore it fails to import.

I had a lot of fields to reorder, which was fiddly and slow via the UI so I did it in a code editor, but when I tried to import my changed JSON file, Bee didn't see any changes.

I could work around that problem by editing the view description string and importing that - Bee detected that change and imported the file, which also updated the order of the fields. Then I undo the text change and reimport, the result is that the fields are ordered as I wanted. So, there is that work around available, but the bug still stands.

[EDIT]:
## How to reproduce:

1. Create a view of Content with two fields: nid and title
2. Do a full export at `admin/config/development/configuration/full/export`
3. Modify the view above by changing the order of the two fields in the UI
4. Head to `admin/config/development/configuration/full` and import the zip file you exported in step 1. This should redirect you to the sync UI page to complete the synchronization.

## Actual behavior
You get the message `The uploaded configuration matches the current site configuration exactly. No changes were made.`  The import did not detect the change of order in the fields. The modifications do not appear in the list of changes in the Synchronize UI

## Expected behavior
The changes should show up, allowing you to import the staged modified file into active.

## Additional info
The culprit is in config.admin.inc, line 523. Currently that line has the "not equal" operator. It should use the "not identical" one.

### PR DESCRIPTION
**Pull request by argiepiano**
Fixes https://github.com/backdrop/backdrop-issues/issues/6348

### CODE REVIEW

File: core/modules/config/tests/config.test [hunk 1]
```diff
@@ -539,6 +539,34 @@ class ConfigurationUITest extends BackdropWebTestCase {
     $this->assert(!empty($staging_files), 'Staging directory is not empty.');
   }
 
+  /**
+   * Test comparing configs whose elements are in different order.
+   */
+  public function testConfigOrderChange() {
```
Comment (indigoxela) [inline core/modules/config/tests/config.test:545]: This doesn't have to be public, it could be protected.

Comment (argiepiano) [reply core/modules/config/tests/config.test:545]: It can, but all the other test methods are all public in this class. Why would we want to change that?

Comment (indigoxela) [reply core/modules/config/tests/config.test:545]: I know, method visibility in functional tests is a total mess throughout core. :wink: 

It doesn't change anything substantially, but I belief, they should all be "protected". At least, newly created ones should be. I'll leave it up to you to change, or keep the current visibility. It's really not relevant for the actual bugfix here.

**Comment by quicksketch** [comment]
Merged into 1.x and 1.26.x. Thanks @argiepiano!

**Pull request merged**